Formula/opam@2.rb

```diff
@@ -1,21 +1,12 @@
 class OpamAT2 < Formula
   desc "The OCaml package manager v2.0.0 (beta)"
   homepage "https://opam.ocaml.org"
-  url "https://github.com/ocaml/opam/releases/download/2.0.0-beta3/opam-full-2.0.0-beta3.tar.gz"
-  sha256 "83d61052b29e7c08674852f4d607e5403720c5fd85664c3b340acf9a938fd9e6"
+  url "https://github.com/ocaml/opam/releases/download/2.0.0-beta4/opam-full-2.0.0-beta4.tar.gz"
+  sha256 "3142adf9e2b43c79a0d44298312a38cf19ffe68f65ccd838144d839ae3adcd9f"
   head "https://github.com/ocaml/opam.git"
 
   depends_on "ocaml" => :recommended
-
-  # aspcud has a fairly large buildtime dep tree, and uses gringo,
-  # which requires C++11 and is inconvenient to install pre-10.8
-  if MacOS.version > 10.7
-    depends_on "aspcud" => :recommended
-  else
-    depends_on "aspcud" => :optional
-  end
-
-  needs :cxx11 if build.with? "aspcud"
+  depends_on "glpk" => :build
```

Comment:
I'm not sure if don't want still want to use `aspcud` by default on platforms that have it. @AltGr @avsm opinions ?

Reply:
If opam was compiled with the built-in solver, that will be used by default even if `aspcud` is present. If it was compiled without, `aspcud`, or another solver, will be required at run-time.
Note that master doesn't even need `glpk` anymore, it's vendored in.

Reply:
Thanks for your answer @AltGr but it doesn't really answer the question. Do you think we should compile and use `opam` with the built-in solver or not on on macos ?

Reply:
The built-in solver is much better in normal use-cases, so I think you should prefer it when possible.

Reply:
FWIW the reason I decided to use `glpk` and drop `aspcud` is that the [build instruction of opam](https://github.com/ocaml/opam#compiling-this-repo) seems to require `glpk`, which I understand in turn will [enable the built-in solver](https://github.com/ocaml/opam#integrated-solver), and the whole README seems to me like it is advocating the use of the internal solver (might be wrong though).

We could have some logic to choose either one, but if the built-in one is recommended then I think it's much simpler this way. Would love your input.

Reply:
Thanks @AltGr ! @bobbypriambodo could you please update the PR accordingly.

Reply:
@dbuenzli I'm not sure I understand what you mean, what should I update?